lib/compress.js

```diff
@@ -2582,6 +2598,18 @@ merge(Compressor.prototype, {
 
     OPT(AST_Call, function(self, compressor){
         var exp = self.expression;
+        if (compressor.option("reduce_vars")
+            && exp instanceof AST_SymbolRef) {
+            var def = exp.definition();
+            if (def.fixed) {
+                exp = def.fixed;
+                if (compressor.option("unused")
+                    && def.references.length == 1
+                    && exp instanceof AST_Function) {
```

Comment:
Can you please add a comment above the cloning line?  It's not obvious what's going on there:

https://github.com/mishoo/UglifyJS2/pull/1560/files#diff-230c53e13e3ee367aec7d49470eb42caR269 

It would be a shallow clone, I assume. Might that present problems with multiple references to it? Do you only clone functions returning constants?